_posts/2019-09-18-uuid4-tiny.md

```diff
@@ -0,0 +1,48 @@
+---
+author: Daniel Mita
+layout: post
+title: UUID4::Tiny
+subtitle: Cryptographically secure version 4 UUIDs via the Linux getrandom() system call
+tags: perl uuid cryptography cpan module
+---
+
+[UUID::Tiny][uuid-tiny] is nice, but unfortunately due to its usage of
+rand() the version 4 UUIDs it provides are not cryptographically
+strong. We opted to create a lightweight (core modules only!)
+alternative which uses a strong source of randomness.
+
+Linux kernel v3.17 introduced the system call getrandom() which
+returns a specified number of random bytes. It reads from the same
+source as /dev/urandom by default without requiring the opening of
+a file handle.
```

Comment:
the lack of need for a file handle removes another potential attack against the gathering of the random numbers.  Might be worth sneaking that in there to explain why this is worth doing.

Reply:
I don't know much about how this works, is there something I can reference?

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=c6e9d6f38894798696f23c8084ca7edbf16ee895

The bit about "resiliance against
file descriptor exhaustion attacks" is the thing we're talking about.